### DISCUSSION TITLE
Test Scenarios to compare Bun/Node performance

### DISCUSSION THREAD

**Discussion by ozhanefemeral** [oven-sh/bun#11565] [General]
Hello Bun community.

I am doing academic research on JavaScript runtime performance. I will compare Bun vs Node.js runtimes for different aspects such as "as a package manager" or "as script runtime" or "as a HTTP server".

Right now I am trying to come up with test scenarios like clean install times, concurrent HTTP scenarios or various script tasks such as "batch read 10000 files" or "write 10 1gb files".

[Here is the all list of current aspects and test scenarios I planned (Notion link).](https://bunvsnodejs.notion.site/Test-Scenarios-for-Bun-Node-js-performance-90783790c56c4f7b9c2bab5979adeb82?pvs=4)

What other scenarios should I cover? Cheers!

**Comment by guest271314** [comment]
There are dozens of JavaScript runtimes https://gist.github.com/guest271314/bd292fc33e1b30dede0643a283fadc6a. 

Bun does not support HTTP/2. 

Neither Bun nor Node.js support Import Maps the last time I checked. 

The last time I tested Bun was faster than Node.js as a Native Messaging host https://gist.github.com/guest271314/fb8969a956a873e495b890a42ac0d731. I've since changed the code for both runtimes.

Some comparisons for all JavaScript runtimes, not dealing with "performance", just in general, that is, what I test for

- WHATWG Streams implementation
- WHATWG Fetch implementation, including `fetch()` for `file:` protocol support
- Import Maps
- TCP socket capabilities
- Capability to serve one upload stream to a different request
- HTTP/2
- WebTransport
- WebAssembly
- Web Cryptography API
- Reading STDIN, writing to STDOUT, handling STDERR
- Ecmascript Module behaviour for specifiers and dynamic `import()`

**Reply by ozhanefemeral** [reply]
@guest271314 Thank you for your comment! I was away lately so it took me some time to respond, sorry about that.

Indeed, your comparisons are not for performance but maybe "Streaming data" could be a valid test scenario. Calculating response times when streaming some file etc.

Also thank you so much for sharing your test results.
